### ISSUE TITLE
Questions about the data-preprocessing

### ISSUE THREAD

**Issue by qijimrc** [PlusLabNLP/AESOP#1]
Hi Jiao Sun, thank you very much for your outstanding work on the paper and code. There were some problems that I met during the use of this code, and I'm sincerely looking forward to some answers.

1. Is the parsing trees in target files of the training set for both the ParaNMT and QQPPos datasets generated with the raw target files instead of additional annotated exemplar files？

2. In the file "data-preprocessing.py", why the logic of pre-processing raw data (ParaNMT/QQPPos) into the specific format files do not match the actual contents and file names in the "raw-data" directory? According to the actual contents in the "raw-data" directory, it seems that the files "ref.txt" and "tgt.txt" should refer to the target and exemplar files, respectively, for both the validation set and test set, but the code has been written as:

```
if signal != "test":
            src, tgt = f"{input_dir}/{signal}/src.txt", f"{input_dir}/{signal}/tgt.txt"
        else:
            src, tgt = f"{input_dir}/{signal}/src.txt", f"{input_dir}/{signal}/ref.txt"
```

3. In the file "data-preprocessing.py", why does the format of generating target sequences not match the format claimed in the paper? Based on the following code at line 41 and line 58, the outputs should be the original target text sequences instead of "`target syntactic parse` <sep> `paraphrase`".

```
  output_tgt.write(f"{tgt_lines[i]}\n")
```

4. Is there any difference between the usages of "exemplar" and "non-exemplar" for generating data samples? According to the "src" and "tgt" defined at line 78 (there is already a confusion mentioned in question 1 above), the "tgt_pure_parses" should be the same between these two approaches.

5. What data source did the "repe_statistics" file in the "processed-data" directory generate from?

**Comment by sunjiao123sun** [comment]
Hi @qijimrc thanks for opening this issue, as there are quite some questions and I have been a bit busy recently, do you mind if I answer them after NAACL deadline? Or you are facing some imperative deadline too that requires AESOP.

**Comment by sunjiao123sun** [comment]
Hi @qijimrc ; 
1. during training, we can directly use the target paraphrases to get the target syntax tree -- you do not need exemplar sentences, and I believe previous work does not have corresponding exemplar sentences for the training set too. The function of exemplar sentences is to avoid leaking the target paraphrases, which only matter during inference.

2/3/4. not sure if I get what you mean, but feel free to customize our code to meet your needs.

5. the repe_statistics are generated using the training data from both QQPPos and ParaNMT. Basically, we are calculating the occurrence of each syntactic combination in those two data, which will be used for the retrieval part in AESOP. 

I hope this helps!
